crates/sncast/src/starknet_commands/account/deploy.rs

```diff
@@ -29,50 +30,40 @@ pub struct Deploy {
     /// Max fee for the transaction
     #[clap(short, long)]
     pub max_fee: Option<FieldElement>,
-
-    /// Custom open zeppelin contract class hash of declared contract
-    #[clap(short, long)]
-    pub class_hash: Option<FieldElement>,
 }
 
 #[allow(clippy::too_many_arguments)]
 pub async fn deploy(
     provider: &JsonRpcClient<HttpTransport>,
     accounts_file: Utf8PathBuf,
-    name: String,
+    deploy_args: Deploy,
     chain_id: FieldElement,
-    max_fee: Option<FieldElement>,
     wait_config: WaitForTx,
-    class_hash: Option<FieldElement>,
+    account: &str,
     keystore_path: Option<Utf8PathBuf>,
-    account_path: Option<Utf8PathBuf>,
 ) -> Result<InvokeResponse> {
     if let Some(keystore_path_) = keystore_path {
-        let account_path_ = account_path
-            .context("Argument `--account` must be passed and be a path when using `--keystore`")?;
-
```

Comment:
correct me if I'm wrong, but the plan is that we no longer need this, becaause we will require the `type` flag, right? If yes that perhaps this PR should include this change (currently the only option being oz)?

Reply:
This logic is still persisted but in the function [check_account_and_keystore_files_exist](https://github.com/foundry-rs/starknet-foundry/pull/2101/files#diff-746993eb163f47a361248767cf9e32a0e39151785fa5f5ff2cdef29ba9e50a72L630)

Reply:
ah ok, so it is just incorporated to deplot_from_keystore function, thanks for clarification :+1: